core/src/main/java/com/linecorp/armeria/common/logging/ContentPreviewer.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.logging;
+
+import java.nio.charset.Charset;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.internal.ArmeriaHttpUtil;
+
+import io.netty.buffer.ByteBuf;
+
+public interface ContentPreviewer {
+    void onData(HttpData data);
+
+    String produce();
+
+    default void onHeaders(HttpHeaders headers) {}
+
+    ContentPreviewer DISABLED = new ContentPreviewer() {
```

Comment:
How about extracting the implementation into a package-local class such as `NoopContentPreviewer` so that this interface does not have a static field? e.g.

```java
static ContentPreviewer disabled() {
   return NoopContentPreviewer.INSTANCE;
}
```

because we have factory methods already in this interface.